app/src/main/java/bean/pwr/imskamieskiego/GUI/InfoSheet.java

```diff
@@ -0,0 +1,181 @@
+import android.support.design.widget.BottomSheetBehavior;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.time.chrono.HijrahChronology;
+
+import bean.pwr.imskamieskiego.R;
+import bean.pwr.imskamieskiego.model.map.Location;
+import bean.pwr.imskamieskiego.model.map.LocationFactory;
+
+public class InfoSheet {
+    private ConstraintLayout layoutInfoSheet;
+    private Button guideToButton;
+    private TextView placeName;
+    private TextView placeInfo;
+    private BottomSheetBehavior sheetBehavior;
+    private ImageButton expandSheetButton;
+    private ImageView pinButton;
+    private Activity parent;
+    private InfoSheetListener listener;
+    public static int COLLAPSED = BottomSheetBehavior.STATE_COLLAPSED;
+    public static int EXPANDED = BottomSheetBehavior.STATE_EXPANDED;
+    public static int HIDDEN = BottomSheetBehavior.STATE_HIDDEN;
+
+
+
+
+
+    public void setListener(InfoSheetListener listener) {
+        this.listener = listener;
+    }
+
+    public InfoSheet(Activity parent) {
+        this.listener = null;
+        this.parent = parent;
+
+
+
+        layoutInfoSheet = parent.findViewById(R.id.info_sheet_layout);
+        guideToButton = parent.findViewById(R.id.guide_to_button);
+        expandSheetButton = parent.findViewById(R.id.info_button);
+        placeName = parent.findViewById(R.id.place_name);
+        placeInfo = parent.findViewById(R.id.place_info);
+
+
+        guideToButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                listener.onGuideToButtonClicked();
+
+            }
+        });
+        pinButton = parent.findViewById(R.id.pin_button);
+        pinButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Location location = LocationFactory.create("Banana", "Żyrafafafa wysoka jest jak szafa");
+                showInfoSheet(location);
+            }
+        });
+
+
+        expandSheetButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                toggleBottomSheet();
+            }
+        });
+
+        sheetBehavior = BottomSheetBehavior.from(layoutInfoSheet);
+        sheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+        sheetBehavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
+
+
+            @Override
+            public void onStateChanged(@NonNull View infoSheet, int newState) {
+                switch (newState) {
+                    case BottomSheetBehavior.STATE_HIDDEN:
+                        layoutInfoSheet.setVisibility(View.GONE);
+                        Log.i("Bottom sheet", "hidden");
+                        listener.onSheetHidden();
```

Comment:
W sumie to o tym zapomniałem. Za każdym razem, jak wywołujesz metodę z listenera, musisz sprawdzić, czy nie jest on nullem. Przykład:
```
if(listener!=null){
    listener.onSheetHidden();
}
```
W innym przypadku, jeżeli wydarzy się coś, co uruchamia listener, a ten nie został "zarejestrowany", będzie rzucać NullPointerami na prawo i lewo.